packages/teleterm/src/ui/ModalsHost/modals/UserJobRole/UserJobRole.tsx

```diff
@@ -0,0 +1,144 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useRef, useState } from 'react';
+import styled from 'styled-components';
+import {
+  ButtonIcon,
+  ButtonPrimary,
+  ButtonSecondary,
+  Input,
+  Text,
+} from 'design';
+import DialogConfirmation, {
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+} from 'design/DialogConfirmation';
+import { Close } from 'design/Icon';
+import { RadioGroup } from 'design/RadioGroup';
+
+interface UserJobRoleProps {
+  onCancel(): void;
+
+  onSend(jobRole: string): void;
+}
+
+const JOB_OPTIONS = [
+  'Software Engineer',
+  'Support Engineer',
+  'DevOps Engineer',
+  'Solutions Architect',
+  'System Administrator',
+];
+
+const OTHER_JOB_ROLE = 'Other';
+
+export function UserJobRole(props: UserJobRoleProps) {
+  const inputRef = useRef<HTMLInputElement>();
+  const [jobRole, setJobRole] = useState<string | null>(null);
+  const [otherJobRole, setOtherJobRole] = useState('');
+
+  const jobRoleToSubmit = jobRole === OTHER_JOB_ROLE ? otherJobRole : jobRole;
+
+  function handleRadioGroupChange(selectedRole: string): void {
+    setJobRole(selectedRole);
+    if (selectedRole === OTHER_JOB_ROLE) {
+      inputRef.current.focus();
+    }
+  }
+
+  function send(): void {
+    props.onSend(jobRoleToSubmit);
+  }
+
+  return (
+    <DialogConfirmation
+      open={true}
+      onClose={props.onCancel}
+      dialogCss={() => ({
+        maxWidth: '400px',
+        width: '100%',
+      })}
+    >
+      <form
+        onSubmit={e => {
+          e.preventDefault();
+          send();
+        }}
+      >
+        <DialogHeader
+          justifyContent="space-between"
+          mb={1}
+          alignItems="baseline"
+        >
+          <Text typography="h4" bold>
+            What describes your current job role best?
+          </Text>
+          <ButtonIcon
+            type="button"
+            onClick={props.onCancel}
+            color="text.secondary"
+          >
+            <Close fontSize={5} />
+          </ButtonIcon>
+        </DialogHeader>
+        <DialogContent mb={3}>
+          <RadioGroup
+            autoFocus={true}
+            name="jobRole"
+            options={[...JOB_OPTIONS, OTHER_JOB_ROLE]}
+            value={jobRole}
+            onChange={handleRadioGroupChange}
+          />
+          <DarkInput
+            ref={inputRef}
+            value={otherJobRole}
+            onChange={e => setOtherJobRole(e.target.value)}
+            onFocus={() => setJobRole(OTHER_JOB_ROLE)}
```

Comment:
I just realized that my suggestion was not so great because now if you choose one of the predefined options, you cannot really tab through to the submit button without selecting "Other".

How about this:

```suggestion
            onChange={e => {
              setJobRole(OTHER_JOB_ROLE);
              setOtherJobRole(e.target.value);
            }}
```

Reply:
> WDYT about changing the option also when the user clicks on the input?

Great idea, this seems to cover pretty much all the ways in which you could use a keyboard or a mouse to interact with the form.

Reply:
Done :)